File: fix_hoisting.py
import os
import re

file_path = 'd:/OneDrive/Bridge stuff/bridge_traffic_deploy/app.js'
with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

# 1. Find and move BMS_CODE_LOOKUPS
bms_start = content.find('const BMS_CODE_LOOKUPS = {')
if bms_start != -1:
    # Find the end of this object definition
    # It ends with '};\n' right before '// Helper for code lookup'
    bms_end = content.find('// Helper for code lookup', bms_start)
    if bms_end != -1:
        # Move back to include '};\n'
        bms_block = content[bms_start:bms_end]
        content = content[:bms_start] + content[bms_end:]
        
        # Insert it after helpers section
        insert_pos = content.find('// ===========================================\n//  CONTROLLER & STATE')
        content = content[:insert_pos] + bms_block + '\n' + content[insert_pos:]

# 2. Find and move selectedMapBridge
smb_str = 'let selectedMapBridge = null;'
if smb_str in content:
    content = content.replace(smb_str + '\n', '')
    content = content.replace(smb_str + '\r\n', '')
    
    insert_pos = content.find('let chartInstances = {};')
    content = content[:insert_pos] + smb_str + '\n' + content[insert_pos:]

with open(file_path, 'w', encoding='utf-8') as f:
    f.write(content)

print("Hoisting fixes applied.")

File: fix_hoisting.js
const fs = require('fs');
const path = require('path');

const filePath = path.join('d:/OneDrive/Bridge stuff/bridge_traffic_deploy', 'app.js');
let content = fs.readFileSync(filePath, 'utf8');

// Find and extract BMS_CODE_LOOKUPS
const bmsRegex = /const\s+BMS_CODE_LOOKUPS\s*=\s*\{[\s\S]*?\n\};\n/g;
let bmsMatch = bmsRegex.exec(content);
if (bmsMatch) {
    let bmsCode = bmsMatch[0];
    content = content.replace(bmsCode, '');
    // Insert after //  HELPERS & CONSTANTS
    content = content.replace('//  HELPERS & CONSTANTS\n// ===========================================\n', '//  HELPERS & CONSTANTS\n// ===========================================\n' + bmsCode);
} else {
    console.log("Could not find BMS_CODE_LOOKUPS");
}

// Find and extract selectedMapBridge
const smbRegex = /let\s+selectedMapBridge\s*=\s*null;\n/g;
let smbMatch = smbRegex.exec(content);
if (smbMatch) {
    let smbCode = smbMatch[0];
    content = content.replace(smbCode, '');
    // Insert after //  CONTROLLER & STATE
    content = content.replace('//  CONTROLLER & STATE\n// ===========================================\n', '//  CONTROLLER & STATE\n// ===========================================\n' + smbCode);
} else {
    console.log("Could not find selectedMapBridge");
}

fs.writeFileSync(filePath, content, 'utf8');
console.log("Fixes applied.");
